pkg/resource/compare/utils.go

```diff
@@ -0,0 +1,41 @@
+package compare
+
+import (
+	"github.com/RHsyseng/operator-utils/pkg/resource"
+	"reflect"
+)
+
+type mapBuilder struct {
+	resourceMap map[reflect.Type][]resource.KubernetesResource
+}
+
+func NewMapBuilder() *mapBuilder {
+	this := &mapBuilder{resourceMap: make(map[reflect.Type][]resource.KubernetesResource)}
+	return this
+}
+
+func (this *mapBuilder) Map() map[reflect.Type][]resource.KubernetesResource {
+	return this.resourceMap
```

Comment:
why not just make `resourceMap` public? i.e. `ResourceMap` or just `Map`?
If not, this function should be renamed to `ResourceMap` as it is not _mapping_ anything but returning the _resourceMap_